src/libsyntax/feature_gate.rs

```diff
@@ -108,124 +108,227 @@ macro_rules! declare_features {
     (active, box_syntax, "1.0.0", Some(49733), None),
-    (active, unboxed_closures, "1.0.0", Some(29625), None),
 
-    (active, fundamental, "1.0.0", Some(29635), None),
+    // Allows using `#[main]` to replace the entrypoint `#[lang = "start"]` calls.
     (active, main, "1.0.0", Some(29634), None),
-    (active, needs_allocator, "1.4.0", Some(27389), None),
-    (active, on_unimplemented, "1.0.0", Some(29628), None),
-    (active, plugin, "1.0.0", Some(29597), None),
-    (active, simd_ffi, "1.0.0", Some(27731), None),
+
+    // Allows using `#[start]` on a function indicating that it is the program entrypoint.
     (active, start, "1.0.0", Some(29633), None),
-    (active, structural_match, "1.8.0", Some(31434), None),
-    (active, panic_runtime, "1.10.0", Some(32837), None),
-    (active, needs_panic_runtime, "1.10.0", Some(32837), None),
 
-    // Features specific to OIBIT (auto traits)
-    (active, optin_builtin_traits, "1.0.0", Some(13231), None),
+    // Allows using the `#[fundamental]` attribute.
+    (active, fundamental, "1.0.0", Some(29635), None),
 
-    // Allows `#[staged_api]`.
-    //
-    // rustc internal
-    (active, staged_api, "1.0.0", None, None),
+    // Allows using the `rust-call` ABI.
+    (active, unboxed_closures, "1.0.0", Some(29625), None),
 
-    // Allows `#![no_core]`.
-    (active, no_core, "1.3.0", Some(29639), None),
+    // Allows using the `#[linkage = ".."]` attribute.
+    (active, linkage, "1.0.0", Some(29603), None),
+
+    // Allows features specific to OIBIT (auto traits).
+    (active, optin_builtin_traits, "1.0.0", Some(13231), None),
 
-    // Allows the use of `box` in patterns (RFC 469).
+    // Allows using `box` in patterns (RFC 469).
     (active, box_patterns, "1.0.0", Some(29641), None),
 
-    // Allows the use of the `unsafe_destructor_blind_to_params` attribute (RFC 1238).
+    // no tracking issue START
+
+    // Allows using `#[prelude_import]` on glob `use` items.
+    (active, prelude_import, "1.2.0", None, None),
+
+    // no tracking issue END
+
+    // Allows using `#[unsafe_destructor_blind_to_params]` (RFC 1238).
     (active, dropck_parametricity, "1.3.0", Some(28498), None),
 
+    // FIXME(Centril): Investigate whether this gate actually has any effect.
+    (active, needs_allocator, "1.4.0", Some(27389), None),
+
+    // no tracking issue START
+
+    // Allows using `#[omit_gdb_pretty_printer_section]`.
+    (active, omit_gdb_pretty_printer_section, "1.5.0", None, None),
+
+    // Allows using the `vectorcall` ABI.
+    (active, abi_vectorcall, "1.7.0", None, None),
+
+    // no tracking issue END
+
     // Allows using the `may_dangle` attribute (RFC 1327).
     (active, dropck_eyepatch, "1.10.0", Some(34761), None),
 
-    // Allows the use of custom attributes (RFC 572).
-    (active, custom_attribute, "1.0.0", Some(29642), None),
+    // Allows using `#[structural_match]` which indicates that a type is structurally matchable.
```

Comment:
```suggestion
    // Allows using `#[structural_match]` which indicates that constants of that type
    // may be used in patterns
```

Reply:
I believe `#[structural_match]` does / will do more than this; See: https://github.com/rust-lang/rfcs/blob/master/text/2000-const-generics.md#structural-equality wherein `#[structural_match]` is required for const generics; cc @varkor cc https://github.com/rust-lang/rust/issues/60286